### PR TITLE
Fix addEntry for reference relation widget 

### PR DESCRIPTION
**Pull request by troopa81**
## Description

When adding an entry from the reference relation widget, the widget was using the referencing field index and not the referenced one, so it fails to update correctly the combo box after the form pop up. 

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Comment by nyalldawson** [comment]
Does this need backporting?

**Comment by troopa81** [comment]
Yes, I just add the labels

File: src/gui/editorwidgets/qgsrelationreferencewidget.cpp [hunk 1]
```diff
@@ -208,7 +208,6 @@ void QgsRelationReferenceWidget::setRelation( const QgsRelation &relation, bool
       mComboBox->setIdentifierField( mReferencedField );
 
     mReferencedFieldIdx = mReferencedLayer->fields().lookupField( relation.fieldPairs().at( 0 ).second );
-    mReferencingFieldIdx = mReferencingLayer->fields().lookupField( relation.fieldPairs().at( 0 ).first );
```
Comment (m-kuhn) [inline src/gui/editorwidgets/qgsrelationreferencewidget.cpp]: This should still be initialized I guess?

Comment (troopa81) [reply src/gui/editorwidgets/qgsrelationreferencewidget.cpp]: actually, the variable has no use anymore. I just remove it 4ee2a53

**Review by rldhont — APPROVED** [review]

**Pull request merged**